vortex-ffi/src/file.rs

```diff
@@ -213,6 +226,47 @@ pub unsafe extern "C-unwind" fn vx_file_reader_free(file: *mut vx_file_reader) {
     drop(Box::from_raw(file));
 }
 
+/// Given an path to a (non-existent file) and an array stream, this function
+/// writes the stream to the file.
```

Comment:
I don't understand this API. You say it opens the file and writes the stream into it. But it returns a handle to a stream writer. Shouldn't it return nothing and just error on failure? Consuming the whole stream?

Reply:
I think its a poor comment, this takes an array stream and begins writing batch to the file, when you push a batch in once side its written out.

We cannot consume the whole stream otherwise the whole file would need to be buffer in memory